src/tree_builder/actions.rs

```diff
@@ -747,7 +747,7 @@ impl<Handle, Sink> TreeBuilderActions<Handle>
     fn insert_element(&mut self, push: PushFlag, ns: Namespace, name: LocalName, attrs: Vec<Attribute>)
             -> Handle {
         declare_tag_set!(form_associatable =
-            "button" "fieldset" "input" "keygen" "label"
+            "button" "fieldset" "input" "keygen" "label" "legend"
```

Comment:
https://html.spec.whatwg.org/multipage/forms.html#form-associated-element seems to contradict this.